File: leetcode/01_99/26__Remove_Duplicates_from_Sorted_Array.java
class Solution {
    public int removeDuplicates(int[] nums) {
        if (nums.length == 0) {
            return 0;
        }
        int l = 1;
        for (int r = 1; r < nums.length; r++) {
            if (nums[r] != nums[l - 1]) {
                nums[l] = nums[r];
                l++;
            }
        }
        return l;
    }
}

File: leetcode/01_99/26__Remove_Duplicates_from_Sorted_Array.c
int removeDuplicates(int* nums, int numsSize){
    if (numsSize == 0) {
        return 0;
    }
    int prev = nums[0];
    int l = 1, r = 1;
    for (; r < numsSize; r++) {
        if (nums[r] != prev) {
            nums[l] = nums[r];
            l++;
            prev = nums[r];
        }
    }
    return l;
}